### PR TITLE
qt3: remove

### LINKED ISSUE #90955: Vulnerability roundup 85: qt-3.3.8: 13 advisories [9.8]
[search](https://search.nix.gsc.io/?q=qt&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=qt+in%3Apath&type=Code)

* [ ] [CVE-2017-10904](https://nvd.nist.gov/vuln/detail/CVE-2017-10904) CVSSv3=9.8 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2018-19873](https://nvd.nist.gov/vuln/detail/CVE-2018-19873) CVSSv3=9.8 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2015-1290](https://nvd.nist.gov/vuln/detail/CVE-2015-1290) CVSSv3=8.8 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2018-19870](https://nvd.nist.gov/vuln/detail/CVE-2018-19870) CVSSv3=8.8 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2018-21035](https://nvd.nist.gov/vuln/detail/CVE-2018-21035) CVSSv3=7.5 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2018-19869](https://nvd.nist.gov/vuln/detail/CVE-2018-19869) CVSSv3=6.5 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2018-19871](https://nvd.nist.gov/vuln/detail/CVE-2018-19871) CVSSv3=6.5 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2017-10905](https://nvd.nist.gov/vuln/detail/CVE-2017-10905) CVSSv3=5.3 (nixos-20.03, nixos-unstable)
* [ ] [CVE-2014-0190](https://nvd.nist.gov/vuln/detail/CVE-2014-0190) (nixos-20.03, nixos-unstable)
* [ ] [CVE-2015-0295](https://nvd.nist.gov/vuln/detail/CVE-2015-0295) (nixos-20.03, nixos-unstable)
* [ ] [CVE-2015-1858](https://nvd.nist.gov/vuln/detail/CVE-2015-1858) (nixos-20.03, nixos-unstable)
* [ ] [CVE-2015-1859](https://nvd.nist.gov/vuln/detail/CVE-2015-1859) (nixos-20.03, nixos-unstable)
* [ ] [CVE-2015-1860](https://nvd.nist.gov/vuln/detail/CVE-2015-1860) (nixos-20.03, nixos-unstable)

Scanned versions: nixos-20.03: a84b797b28e; nixos-unstable: 22c98819ccd. May contain false positives.

### LINKED ISSUE #96827: Vulnerability roundup 92: qt-3.3.8: 1 advisory [5.3]
[search](https://search.nix.gsc.io/?q=qt&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=qt+in%3Apath&type=Code)

* [ ] [CVE-2020-17507](https://nvd.nist.gov/vuln/detail/CVE-2020-17507) CVSSv3=5.3 (nixos-20.03, nixos-unstable)

Scanned versions: nixos-20.03: 925ae0dee63; nixos-unstable: c59ea8b8a0e.

### LINKED ISSUE #99841: Vulnerability roundup 93: qt-3.3.8: 1 advisory [7.3]
[search](https://search.nix.gsc.io/?q=qt&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=qt+in%3Apath&type=Code)

* [ ] [CVE-2020-0570](https://nvd.nist.gov/vuln/detail/CVE-2020-0570) CVSSv3=7.3 (nixos-20.03, nixos-20.09, nixos-unstable)

Scanned versions: nixos-20.03: 0d0660fde3b; nixos-20.09: ad3a5d5092e; nixos-unstable: 84d74ae9c9c.

### PR DESCRIPTION
**Pull request by doronbehar**
Not used by any other package, and has vulnerabilities.

Close #99841 , Close #96827, Close #90955.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> Not used by any other package, and has vulnerabilities.

```
anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/tools/graphics/lprof/default.nix:1:1 called without required argument 'qt3', at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/customisation.nix:69:16
```

**Comment by doronbehar** [comment]
> > Not used by any other package, and has vulnerabilities.
> 
> ```
> anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/tools/graphics/lprof/default.nix:1:1 called without required argument 'qt3', at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/customisation.nix:69:16
> ```

Thanks. Removed lprof as well, it was broken since a while now.

**Comment by doronbehar** [comment]
But I see now that taskjuggler also needs qt3... Not sure if to remove it as well, there's a newer version of taskjuggler that doesn't need qt3.

**Comment by doronbehar** [comment]
> But I see now that taskjuggler also needs qt3... Not sure if to remove it as well, there's a newer version of taskjuggler that doesn't need qt3.

I take my words back. The nix files of taskjuggler 2 were not used at all and they only showed up in `git grep` but had no usage from `all-packages.nix`.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4155,8 +4155,6 @@ in
 
   lp_solve = callPackage ../applications/science/math/lp_solve { };
 
-  lprof = callPackage ../tools/graphics/lprof { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:4158]: Can we add aliases for the removal?

Comment (doronbehar) [reply pkgs/top-level/all-packages.nix:4158]: Done.

**Pull request merged**